### PR TITLE
Try: A block-based version of Mayland

### PR DESCRIPTION
**Pull request by kjellr**
This PR is the beginnings of a block-based version of Mayland. 

- The social icons block has colored icons for now — we could add a theme-specific block style to take care of that though. 
- Post meta blocks can't currently be placed inline, so the single line of post meta blocks isn't possible right now. For now I've included only the post date, and filed https://github.com/WordPress/gutenberg/issues/26899. 
- Comments look unstyled — these blocks don't provide styles, but they should. I filed some issues for that yesterday: https://github.com/WordPress/gutenberg/issues/26864 and https://github.com/WordPress/gutenberg/issues/26865.
- There are a number of small spacing/style issues that don't match up exactly with the original theme, but I expect we could sort those out.

## Screenshots

Original|Block-based
---|---
maylanddemo wordpress com_|dotcomthemes test__page_id=12

Original|Block-based
---|---
maylanddemo wordpress com_about_|dotcomthemes test__page_id=2

Original|Block-based
---|---
maylanddemo wordpress com_2019_09_05_overcoming-depth-of-field-limitations_|dotcomthemes test__p=1

### CODE REVIEW

**Review by jffng — COMMENTED** [review]
In general this LGTM. 

I tested this on 9.3 (the latest plugin release as of this review) and the font families are not transformed nor applied.

As of 9.4.0rc1, the font families work but the template parts aren't loading in the site editor: 

Screen Shot 2020-11-17 at 12 30 39 PM

**Comment by MaggieCabrera** [comment]
I do see header and footer on 9.3.0 but not the fonts 😃

**Comment by kjellr** [comment]
Yeah, I think the template parts issue is a 9.4 bug. [Filed here](https://href.li/?https://github.com/WordPress/gutenberg/issues/27072).

**Comment by kjellr** [comment]
This has been dormant for a while, so here's a fresh set of things that it needs: 

- [x] Reformat theme.json to the new spec. 
- [x] Remove [theme slugs in templates](https://href.li/?https://github.com/WordPress/gutenberg/pull/28088).
- [ ] Consider removing the spacing overrides in `style.css`. 
- [ ] Add query pagination. 
- [x] Double check that comments are working. 

cc @scruffian

**Comment by MaggieCabrera** [comment]
I've worked a bit on this theme. One of the things I found was that when adding the Next/Previous links to a single post template I was not able to make the "Previous/Next post" text a link to the post just using the block. The final result of the block will require some extra CSS to make it fit the design of the theme as well (I haven't added it yet but I plan to do so unless there are any objections to that)

Original | Block-Based
---  | ---
Screenshot 2021-02-15 at 14 58 07 | Screenshot 2021-02-15 at 14 57 58

**Comment by scruffian** [comment]
I feel like that's a pretty common pattern so should be added to the editor

**Comment by kjellr** [comment]
Yeah, I agree — that should be a variation for the core block. If we do decide to build it in here, I suggest we do it as a block style maybe?

**Comment by MaggieCabrera** [comment]
> Yeah, I agree — that should be a variation for the core block. If we do decide to build it in here, I suggest we do it as a block style maybe?

I'm not sure if a block style is the best idea here since turning off the block style would make it look not so great. I'd say it's best if it's just styled by the theme (I'm talking in terms of just the font sizes and positioning of the links). Hopefully, the block will allow for more styling in the future too.

Besides, the issue with the "Previous/Next post" not being a link to the post too can't be solved by a block style either. I opened an issue for the block, which is also a very very new block to be using, so I suppose it will only improve with time (and our contributions, ofc): https://github.com/WordPress/gutenberg/issues/29032

**Comment by kjellr** [comment]
If it doesn't exist the way we need it to yet, I'd suggest we just leave that post pagination out of the theme and open a ticket to add it later.

**Comment by MaggieCabrera** [comment]
I added the issue for the post pagination [here](https://github.com/Automattic/themes/issues/3286) and removed the block for the moment

**Comment by MaggieCabrera** [comment]
For reference: [this Gutenberg](https://github.com/WordPress/gutenberg/issues/23228) issue holds the discussion for implementing letter spacing controls on FSE

File: mayland-blocks/functions.php [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+	return esc_url_raw( $fonts_url );
+}
+
+/**
+ * Enqueue scripts and styles.
+ */
+function mayland_blocks_scripts() {
+	// Enqueue front-end styles.
+	wp_enqueue_style( 'mayland-blocks-styles', get_template_directory_uri() . '/style.css' );
```
Comment (scruffian) [inline mayland-blocks/functions.php]: Do we need this line? I thought this file was enqueued by default. Also we could just use `get_stylesheet_uri`...

File: mayland-blocks/functions.php [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+	// Enqueue front-end styles.
+	wp_enqueue_style( 'mayland-blocks-block-styles', get_template_directory_uri() . '/assets/alignments.css' );
```
Comment (scruffian) [inline mayland-blocks/functions.php]: Should we use `get_stylesheet_directory_uri` instead?

File: mayland-blocks/style.css [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+This theme, like WordPress, is licensed under the GPL.
+Use it to make something cool, have fun, and share what you've learned with others.
+*/
+
+
+/* Smooth out the fonts. */
+
+body {
+	-webkit-font-smoothing: antialiased;
+	-moz-osx-font-smoothing: grayscale;
```
Comment (scruffian) [inline mayland-blocks/style.css:23]: This feels like it should be part of Gutenberg

File: mayland-blocks/style.css [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+@media screen and (min-width: 1290px) {
+	.site-header .wp-block-column,
+	.site-footer .wp-block-column {
+		padding-left: var(--wp--custom--margin--horizontal);
+		padding-right: var(--wp--custom--margin--horizontal);
+	}
+}
+
+/* Adjust heading letter spacing. */
```
Comment (scruffian) [inline mayland-blocks/style.css:63]: We could write this more concisely by grouping the selectors. I also don't think we need the rules for h4, h5 and h6.

**Review by scruffian — APPROVED** [review]
This looks good. I think everything in style.css should be in Gutenberg somehow, so it would be good to have issues for it if there aren't already.

File: mayland-blocks/functions.php (hunk 1 above)
Comment (MaggieCabrera) [reply mayland-blocks/functions.php]: Apparently we do, if I remove it it doesn't load automatically

File: mayland-blocks/style.css (hunk 3 above)
Comment (MaggieCabrera) [reply mayland-blocks/style.css:23]: Not sure if every theme will want to have this on their stylesheet, feels like something we want for this specific font family.

What is under `/* Set editor widths. */` would be something we'd be able to remove once the alignment PR lands most likely. 
The rest of the styles (header spacing and letter spacing) are dependant on https://github.com/WordPress/gutenberg/issues/23294 and https://github.com/WordPress/gutenberg/issues/23228 respectively

**Comment by MaggieCabrera** [comment]
This is a list of the editor issues that are affecting this theme:

- Navigation block: Add a way to control spacing https://github.com/WordPress/gutenberg/issues/23294
- Site Title: A few styling option ideas https://github.com/WordPress/gutenberg/issues/23228
- Rethink FSE alignments https://github.com/WordPress/gutenberg/pull/29335
- [Block: Post Navigation Link] Add the ability to have both the text title and the label be a part of the link https://github.com/WordPress/gutenberg/issues/29032
- Post Comments Form Block: Block should ship with default styles https://github.com/WordPress/gutenberg/issues/26865
- Post Comments Block: Provide default styles https://github.com/WordPress/gutenberg/issues/26864
- Allow post meta to be arranged inline https://github.com/WordPress/gutenberg/issues/26899

File: mayland-blocks/assets/alignments.css [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+	width: var(--wp--custom--width--wide);
+	max-width: 100%;
+	margin-left: auto;
+	margin-right: auto;
+}
+
+.wp-site-blocks .alignfull {
+	transform: translateX(calc(0px - var(--wp--custom--width-horizontal)));
+	width: calc(100% + (2 * var(--wp--custom--width-horizontal)));
+	max-width: calc(100% + (2 * var(--wp--custom--width-horizontal)));
```
Comment (pbking) [inline mayland-blocks/assets/alignments.css]: I believe all of these should be `--wp--custom--margin--horizontal`

Comment (MaggieCabrera) [reply mayland-blocks/assets/alignments.css]: oh yes, should we get the version from TT1 blocks? I remember we amended a few things back there

**Review by danieldudzic — COMMENTED** [review]
I didn't find any blockers for the launch :)

Just a few issues with the frontend styling: 
1. Layout Grid columns are displaying one under another
Screenshots: https://cldup.com/BwY5yx1W2S.png / https://cldup.com/cnI7FwSEtZ.png

2. Button styling issue
Screenshots: https://cldup.com/oHi6rbHij4.png / https://cldup.com/B7j0OfyOyZ.png
3. Contact form missing styling: https://cldup.com/BZUulYROfK.png

File: mayland-blocks/assets/alignments.css (hunk 5 above)
Comment (MaggieCabrera) [reply mayland-blocks/assets/alignments.css]: That didn't work, so I just updated this

**Comment by MaggieCabrera** [comment]
> I didn't find any blockers for the launch :)
> 
> Just a few issues with the frontend styling:

Looks like some of those are Jetpack blocks causing issues. I didn't see problems with the Button Block on my testing, though. 

Screenshot 2021-02-26 at 17 06 44


These issues will be easier to test and report on once the theme is on wpcom.

**Pull request merged**